Makefile

```diff
@@ -84,13 +84,18 @@ test-unit: envtest ## Run the unit tests
 e2e: KIND_CLUSTER_NAME=operator-controller-e2e
 e2e: run test-e2e kind-cluster-cleanup ## Run e2e test suite on local kind cluster
 
-kind-load: kind ## Loads the currently constructed image onto the cluster
+kind-load: kind ## Loads the currently constructed image onto the kind cluster
 	$(KIND) load docker-image $(IMG) --name $(KIND_CLUSTER_NAME)
 
 kind-cluster: kind kind-cluster-cleanup ## Standup a kind cluster
 	$(KIND) create cluster --name ${KIND_CLUSTER_NAME}
 	$(KIND) export kubeconfig --name ${KIND_CLUSTER_NAME}
 
+kind-deploy: export MANIFEST="./operator-controller.yaml"
+kind-deploy: kind manifests kustomize ## Install controller and dependencies onto the kind cluster
+	kubectl kustomize config/default > operator-controller.yaml
+	envsubst '$$CATALOGD_VERSION,$$CERT_MGR_VERSION,$$RUKPAK_VERSION,$$MANIFEST' < scripts/install.tpl.sh | bash -s
```

Comment:
Just curious, if we have this, then do we need `deploy` target. I see that `make kind-deploy` would controller and the dependencies and the latter would only install the controller, but is there a use case where we would like to support just the installation of the controller with (probably) some other version of residual dependencies running on cluster. 

Also, thoughts on having the naming as `kind-deploy-e2e`? Just so that it's easier to interpret the difference between the two.

Reply:
How much of this is theoretical? I'd very much like to limit our Makefile to having as few supported use-cases as possible, for the sake of our own sanity.

I personally use (pretty much exclusively):
- `make build` to build the binaries locally.
- `make run` to rebuild/redeploy everything in a fresh kind cluster
- `make test-unit` (or whatever we call it) to run the unit tests
- `make test-e2e` to run the e2e test

Would it be worthwhile to do a quick contributor survey to see what people _actually_ use and then go from there?

Reply:
https://kubernetes.slack.com/archives/C0181L6JYQ2/p1685127976360479